### PR TITLE
chore: implement token endpoint url

### PR DESCRIPTION
**Pull request by hemahg**
**What this PR does / why we need it**:

> implement token endpoint url

**Which issue(s) this PR fixes** 

> Replace this comment with references to related issues.
> e.g. `fixes #<issue number>`

**Verification steps** 

> Attach a screenshot of the output.
> guidance on how to go the particular story.
> e.g. `point to the story with a text breadcrumb, something like "components > <component name> > <story name>”`

**Is there any work left / what's next** :

> Replace this comment with pending items that need to be covered in this PR or some pending items you are planning to do in the future PR.

**Special notes for your reviewer**:

### CODE REVIEW

File: apps/consoledot-rhosak/src/routes/control-plane/DrawerProvider.tsx [hunk 1]
```diff
@@ -47,6 +47,18 @@ export const DrawerProvider: FunctionComponent = ({ children }) => {
...
+    undefined
+  );
+
+  const { data: ssoProviders } = useSSOProviders();
+
+  useEffect(() => {
+    if (ssoProviders) {
+      setTokenEndPointUrl(ssoProviders?.token_url);
+    }
+  }, [ssoProviders]);
```
[the remark this answers is not in the document]
Comment (hemahg) [reply apps/consoledot-rhosak/src/routes/control-plane/DrawerProvider.tsx]: I have removed useState and useEffect

File: packages/consoledot-api/src/queries/useSSOProvider.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { useQuery } from "@tanstack/react-query";
+import { useConfig } from "@rhoas/app-services-ui-shared";
+import { masQueries } from "../queryKeys";
+import type { SsoProviderAllOf } from "@rhoas/kafka-management-sdk";
+
+export const useSSOProviders = () => {
+  const config = useConfig();
+
+  return useQuery<SsoProviderAllOf>({
```
[the remark this answers is not in the document]
Comment (hemahg) [reply packages/consoledot-api/src/queries/useSSOProvider.ts]: I looked into the SDK, But I am not able to find the function that retrieve the data from sdk

**Review by riccardo-forina — APPROVED** [review]
Did you redo an npm install?

File: packages/consoledot-api/src/queries/useSSOProvider.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      );
+      const providers = await response.json();
+      const { token_url } = providers;
+      const ssoProvider: SsoProvider = { tokenEndPointUrl: token_url };
+      return ssoProvider;
+    },
+  });
+  const tokenEndPointUrl = ssoProviders?.tokenEndPointUrl;
+
+  if (!tokenEndPointUrl) {
```
Comment (hemahg) [inline packages/consoledot-api/src/queries/useSSOProvider.ts:26]: @riccardo-forina, In this line I am trying to check whether the tokenEndPointUrl is present or not and throwing the error, but it's making the application stop running and throw `Token URL is undefined`, could you please assist me with what I am doing wrong here

File: packages/consoledot-api/src/queries/useSSOProvider.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      );
+      const providers = await response.json();
+      const { token_url } = providers;
+      const ssoProvider: SsoProvider = { tokenEndPointUrl: token_url };
+      return ssoProvider;
+    },
+  });
+  const tokenEndPointUrl = ssoProviders?.tokenEndPointUrl;
+  return { tokenEndPointUrl };
+};
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queries/useSSOProvider.ts:26]:
```suggestion
export const useSSOTokenEndpoint = () => {
  const { basePath } = useApiConfiguration();

  return useQuery<string>({
    queryKey: masQueries.tokenEndPointUrl(),
    queryFn: async () => {
      const response = await fetch(
        `${basePath}/api/kafkas_mgmt/v1/sso_providers`
      );
      const provider = await response.json();
      const { token_url } = provider;
      if (token_url === undefined || token_url === '') {
        return Promise.reject("Invalid SSO token URL");
      }
      return token_url;
    },
  });
};
```

This is what I meant in my previous comments.

**Review by riccardo-forina — CHANGES_REQUESTED** [review]

File: packages/consoledot-api/src/queries/useSSOProvider.ts [hunk 5]
```diff
@@ -15,7 +11,8 @@ export const useSSOProvider = () => {
       const response = await fetch(
         `${basePath}/api/kafkas_mgmt/v1/sso_providers`
       );
-      const providers = await response.json();
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const providers: { token_url?: string } = await response.json();
```
Comment (riccardo-forina) [inline packages/consoledot-api/src/queries/useSSOProvider.ts:15]:
```suggestion
      const providers = await response.json<{ token_url?: string }>();
```
This should be a bit better, we tell Typescript the shape of the object returned by that json call, so we can do our defensive checks later without having to shut down eslint.

Comment (riccardo-forina) [reply packages/consoledot-api/src/queries/useSSOProvider.ts:15]: Ah, yes sure, I was convinced `fetch` could take a type argument

File: packages/ui/src/components/KafkaInstanceDrawer/components/KafkaConnectionTabP2.tsx [hunk 6]
```diff
@@ -287,7 +287,7 @@ export const KafkaConnectionTabP2: FunctionComponent<
           {t("connection-tab:sasl_oauthbearer_description")}
         </Text>
         <strong>{t("connection-tab:token_endpoint_url")}</strong>
-        {isKafkaPending ? (
+        {isKafkaPending && tokenEndPointUrl ? (
```
Comment (riccardo-forina) [inline packages/ui/src/components/KafkaInstanceDrawer/components/KafkaConnectionTabP2.tsx:290]: This check doesn't look right?

**Review by riccardo-forina — APPROVED** [review]
Looks great but for the thing I commented, I'm not sure that's right

File: packages/ui/src/components/KafkaInstanceDrawer/components/KafkaConnectionTabP2.tsx [hunk 7]
```diff
@@ -287,7 +287,7 @@ export const KafkaConnectionTabP2: FunctionComponent<
           {t("connection-tab:sasl_oauthbearer_description")}
         </Text>
         <strong>{t("connection-tab:token_endpoint_url")}</strong>
-        {isKafkaPending ? (
+        {isKafkaPending || tokenEndPointUrl === undefined ? (
```
Comment (riccardo-forina) [inline packages/ui/src/components/KafkaInstanceDrawer/components/KafkaConnectionTabP2.tsx:290]:
```suggestion
        {tokenEndPointUrl === undefined ? (
```
I don't think we need to wait for the instance to be ready as this url pointed to SSO.

**Review by riccardo-forina — APPROVED** [review]

**Review by riccardo-forina — APPROVED** [review]

**Pull request merged**